internal/metrics/directory_usage_collector.go

```diff
@@ -0,0 +1,60 @@
+package metrics
+
+import (
+	"github.com/alecthomas/units"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	directoryUsageGbName                = "assisted_installer_directory_usage_gb"
+	directoryUsageGbDescription         = "The amount of the free space in a directory that has been used in Gb"
+	directoryUsagePercentageName        = "assisted_installer_directory_usage_percentage"
```

Comment:
Why used GB and percentage? If we have used and free or used and total (which I think it's easier in this case) I think it's good enough. If % is needed, we can calculate it from PromQL. WDYT?

Reply:
I ended up discovering a problem with this approach anyway and had to make things closer to your suggestion.

Now we gather directory usage and directory free space. 

The usage comes from the file content of the target directory.
The free space comes from the backing storage.